lib/handleRequest.js

```diff
@@ -117,7 +117,13 @@ function preHandlerCallback (err, code) {
 
   var result = this.store.handler(this.request, this.reply)
   if (result && typeof result.then === 'function') {
-    this.reply.send(result)
+    result.then((payload) => {
+      // this is for async functions that
+      // are using reply.send directly
+      if (payload !== undefined) {
```

Comment:
This can be wrong:
```
fastify.get('/get', async (req, reply) => {
  const s = await getStrinOrUndefined()
  return s
})
```

The right check should be check if "send" is already called

Reply:
That will force any async/await user to not use .send() at all. Are we ok with that? I can check if the reply is sending or has been sent, and if that's the case we can avoid sending (ad we are doing right now).

reply.send() could be called later on, and the promise could already be resolved to undefined.

Reply:
> That will force any async/await user to not use .send() at all. Are we ok with that?

This would be super tricky and counterintuitive, I'm 👎 .

Reply:
So, are you ok with this change or you prefer another approach?

Reply:
Is not an easy choice,  your fix will solve the issue, but the example of @allevo could be a real world situation.
Which alternatives do we have?

Reply:
> That will force any async/await user to not use .send() at all. Are we ok with that?

if there's a chance to do that I'm 👍 !

In `fastify` the `send` method can be called only 1 time by design.
I don't understand why we don't move [this line](That will force any async/await user to not use .send() at all. Are we ok with that?) to [here](https://github.com/fastify/fastify/blob/master/lib/reply.js#L32)